File: src/main/java/net/dontdrinkandroot/gitki/wicket/page/configuration/ConfigurationPage.java
package net.dontdrinkandroot.gitki.wicket.page.configuration;

import net.dontdrinkandroot.gitki.model.Role;
import net.dontdrinkandroot.gitki.service.configuration.ConfigurationService;
import net.dontdrinkandroot.gitki.service.git.GitService;
import net.dontdrinkandroot.gitki.wicket.page.DecoratorPage;
import org.apache.wicket.authroles.authorization.strategies.role.annotations.AuthorizeInstantiation;
import org.apache.wicket.markup.html.basic.Label;
import org.apache.wicket.model.IModel;
import org.apache.wicket.model.Model;
import org.apache.wicket.spring.injection.annot.SpringBean;
import org.springframework.core.env.Environment;

/**
 * @author Philip Washington Sorst <philip@sorst.net>
 */
@AuthorizeInstantiation(Role.Constants.ADMIN)
public class ConfigurationPage extends DecoratorPage
{
    @SpringBean
    private GitService gitService;

    @SpringBean
    private Environment environment;

    @SpringBean
    private ConfigurationService configurationService;

    @Override
    protected IModel<String> createTitleModel()
    {
        return Model.of("Configuration");
    }

    @Override
    protected void onInitialize()
    {
        super.onInitialize();
        this.add(new Label("repositoryPath", this.gitService.getRepositoryPath().toString()));
        this.add(new Label(
                "anonymousBrowsingEnabled",
                Boolean.toString(this.configurationService.isAnonymousBrowsingEnabled())
        ));
        this.add(new Label("defaultProfiles", String.join(",", this.environment.getDefaultProfiles())));
        this.add(new Label("activeProfiles", String.join(",", this.environment.getActiveProfiles())));
    }
}

File: src/main/kotlin/net/dontdrinkandroot/gitki/wicket/page/admin/ConfigurationPage.kt
package net.dontdrinkandroot.gitki.wicket.page.admin

import net.dontdrinkandroot.gitki.model.Role
import net.dontdrinkandroot.gitki.service.configuration.ConfigurationService
import net.dontdrinkandroot.gitki.service.git.GitService
import net.dontdrinkandroot.gitki.wicket.page.DecoratorPage
import net.dontdrinkandroot.gitki.wicket.security.Instantiate
import net.dontdrinkandroot.wicket.kmodel.ValueKModel
import org.apache.wicket.markup.html.basic.Label
import org.apache.wicket.request.mapper.parameter.PageParameters
import org.apache.wicket.spring.injection.annot.SpringBean
import org.springframework.core.env.Environment

@Instantiate(Role.ADMIN)
class ConfigurationPage(parameters: PageParameters) : DecoratorPage<Void>(parameters) {

    @SpringBean
    private lateinit var gitService: GitService

    @SpringBean
    private lateinit var environment: Environment

    @SpringBean
    private lateinit var configurationService: ConfigurationService

    override fun createTitleModel() = ValueKModel("Configuration")

    override fun onInitialize() {
        super.onInitialize()
        this.add(Label("repositoryPath", gitService.repositoryPath.toString()))
        this.add(
            Label(
                "anonymousBrowsingEnabled",
                configurationService.isAnonymousBrowsingEnabled.toString()
            )
        )
        this.add(Label("defaultProfiles", java.lang.String.join(",", *environment.defaultProfiles)))
        this.add(Label("activeProfiles", java.lang.String.join(",", *environment.activeProfiles)))
    }
}